### PR TITLE
Add example for IndexLayerClient

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
Add usage example for IndexLayerClient

Resolves: OLPEDGE-1337
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: docs/examples/nodejs-read-index-layer.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+   const olpClientSettings = new OlpClientSettings({
+     environment:
+       "here | here-dev | here-cn | here-cn-dev | http://YourCustomEnvironment",
+     getToken: () => userAuth.getToken()
+   });
+   ```
+
+## <a name="create-IndexlayerClient"></a>Create IndexLayerClient`
+
+You can use the `IndexLayerClient` object to request any data and partition metadata from a [index layer](https://developer.here.com/olp/documentation/data-user-guide/portal/layers/creating-index-layer.html).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-index-layer.md]:
```suggestion
You can use the `IndexLayerClient` object to request any data and partition metadata from an [index layer](https://developer.here.com/olp/documentation/data-user-guide/portal/layers/layers.html#index-layers).
```

File: docs/examples/nodejs-read-index-layer.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+- Timestamp
+
+To get partition metadata from an index layer:
+
+1. Create the `IndexLayerClient` object.
+   For instructions, see [Create IndexLayerClient](#create-IndexlayerClient).
+
+2. Create the `IndexQueryRequest` object with the `RSQL query` to retrieve partitions that match the query and `huge` boolean parameter setting `true` if query is huge. More information could be find [Get Data from an Index Layer](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/getting-data-index.html).
+
+    > Note: Default value for the huge parameter is `false` and it is optional.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-index-layer.md]:
```suggestion
    > Note:  The `huge` parameter is optional and its default value is `false`.
```

File: docs/examples/nodejs-read-index-layer.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+- Checksum
+- Metadata
+- Timestamp
+
+To get partition metadata from an index layer:
+
+1. Create the `IndexLayerClient` object.
+   For instructions, see [Create IndexLayerClient](#create-IndexlayerClient).
+
+2. Create the `IndexQueryRequest` object with the `RSQL query` to retrieve partitions that match the query and `huge` boolean parameter setting `true` if query is huge. More information could be find [Get Data from an Index Layer](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/getting-data-index.html).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-index-layer.md]:
```suggestion
2. Create the `IndexQueryRequest` object with the RSQL query string and, if the query string is huge, set the `huge` boolean parameter to `true` .
```
Why do we need a reference to the similar instruction? Is there anything in the referenced instruction that should be mentioned here?

File: docs/examples/nodejs-read-index-layer.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+## Get Data from a Index Layer
+
+To get data from the index layer:
+
+1. Create the `IndexLayerClient` object.
+   For instructions, see [Create IndexLayerClient](#create-IndexlayerClient).
+
+2. Call the `GetData` method with the data model of partition metadata. To get data model could be used method `getPartitions` what returns the list of data models. The mandatory property in data model is `id` what is data handle in the nutshell.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-index-layer.md]:
```suggestion
2. Call the `GetData` method with the data model that contains the partition ID (also used as the data handle). 

  >Note: You can find the data model in the partition metadata. For instructions, see [Get Partition Metadata from an Index Layer](#get-partition-metadata-from-an-index-layer).
```

File: docs/examples/nodejs-read-index-layer.md (hunk 1 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-index-layer.md]: fixed

File: docs/examples/nodejs-read-index-layer.md (hunk 2 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-index-layer.md]: Fixed

File: docs/examples/nodejs-read-index-layer.md (hunk 3 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-index-layer.md]: Fixed

File: docs/examples/nodejs-read-index-layer.md (hunk 4 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-index-layer.md]: Fixed. It is not Partition ID but just ID property with dataHandle value stored there.

**Pull request merged**